cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -170,9 +170,7 @@ public static void validateConfig(final SecureStore secureStore,
       LsRemoteCommand cmd =
           createCommand(git::lsRemote, sourceControlConfig,
               credentialsProvider).setRemote(
-                  config.getLink())
-              .setHeads(true)
```

Comment:
Dont we need `setHeads(true)` for `refs.get(Constants.HEAD)` to work ?

Reply:
There is a unit test named `testValidateNullBranchName` to verify HEAD is fetched correctly.